quickjs.c

```diff
@@ -46026,6 +46048,99 @@ static JSValue js_map_iterator_next(JSContext *ctx, JSValue this_val,
     }
 }
 
+static JSValue js_map_read(BCReaderState *s, int magic)
+{
+    JSContext *ctx = s->ctx;
+    JSValue obj, rv, argv[2];
+    uint32_t i, prop_count;
+
+    argv[0] = JS_UNDEFINED;
+    argv[1] = JS_UNDEFINED;
+    obj = js_map_constructor(ctx, JS_UNDEFINED, 0, NULL, magic);
+    if (JS_IsException(obj))
+        return JS_EXCEPTION;
+    if (BC_add_object_ref(s, obj))
+        goto fail;
+    if (bc_get_leb128(s, &prop_count))
+        goto fail;
+    for(i = 0; i < prop_count; i++) {
+        argv[0] = JS_ReadObjectRec(s);
+        if (JS_IsException(argv[0]))
+            goto fail;
+        if (!(magic & MAGIC_SET)) {
+            argv[1] = JS_ReadObjectRec(s);
+            if (JS_IsException(argv[1]))
+                goto fail;
+        }
+        rv = js_map_set(ctx, obj, countof(argv), argv, magic);
+        if (JS_IsException(rv))
+            goto fail;
+        JS_FreeValue(ctx, rv);
+        JS_FreeValue(ctx, argv[0]);
+        JS_FreeValue(ctx, argv[1]);
+        argv[0] = JS_UNDEFINED;
```

Comment:
Is this necessary since yoy are about to return?

Reply:
Yes, because it's inside a loop, otherwise we can end up double-freeing argv[1] on the next iteration.

(And I'm clearing argv[0] for clarity's sake, otherwise a casual reader will stop and think: "Why one and not the other?")

Reply:
Great!